buf/registry/plugin/v1beta1/plugin.proto

```diff
@@ -0,0 +1,137 @@
+
+message Plugin {
+  option (buf.registry.priv.extension.v1beta1.message).response_only = true;
+
+  // The id of the Plugin.
+  string id = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Plugin was created on the BSR.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The last time the Plugin was updated on the BSR.
+  google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
+  // The name of the Plugin.
+  //
+  // Unique within a given Organization.
+  string name = 4 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 100
+  }];
+  // The id of the Organization that owns the Plugin.
+  string owner_id = 5 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The Plugin's visibility, either public or private.
+  PluginVisibility visibility = 6 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin state, either active or deprecated.
+  PluginState state = 7 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin type.
+  PluginType type = 8 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The configurable short description of the Plugin.
+  string description = 9 [(buf.validate.field).string.max_len = 350];
+  // The configurable source URL of the Plugin.
+  string source_url = 10 [
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
+  ];
+  // Managed indicates whether a Plugin is managed by the BSR.
+  //
+  // Managed Plugins are automatically updated and should not be updated by users because they will
+  // be overwritten.
+  bool managed = 11;
+  // The collections the Plugin belongs to.
+  repeated Collection collections = 12;
+}
+
+// The visibility of a Plugin, currently either public or private.
+enum PluginVisibility {
+  PLUGIN_VISIBILITY_UNSPECIFIED = 0;
+  // PLUGIN_VISIBILITY_PUBLIC says that the plugin is publicly available.
+  PLUGIN_VISIBILITY_PUBLIC = 1;
+  // PLUGIN_VISIBILITY_PRIVATE says that the plugin is private.
+  PLUGIN_VISIBILITY_PRIVATE = 2;
+}
+
+// The state of a Plugin, currently either active or deprecated.
+enum PluginState {
+  PLUGIN_STATE_UNSPECIFIED = 0;
+  // PLUGIN_STATE_ACTIVE says that the Plugin is currently active.
+  PLUGIN_STATE_ACTIVE = 1;
+  // PLUGIN_STATE_DEPRECATED says that the Plugin has been deprecated and should no longer be used.
+  PLUGIN_STATE_DEPRECATED = 2;
+}
+
+// The type of the Plugin.
+enum PluginType {
+  PLUGIN_TYPE_UNSPECIFIED = 0;
+  // PLUGIN_TYPE_CHECK says that the Plugin is a check plugin.
+  PLUGIN_TYPE_CHECK = 1;
```

Comment:
There was a comment about `PLUGIN_TYPE_PROTOC` that was deleted from the current view, because `PLUGIN_TYPE_PROTOC` was removed without us concluding the discussion around whether or not it should be included. I was confused when I couldn't find the discussion.

The answer to whether `PLUGIN_TYPE_PROTOC` should be included depends on a few areas:

- Can you map version/revision to label/commit?
- What does it mean to request a download of a protoc plugin? What error is returned otherwise?
- How do you explain in the docs that you cannot upload a plugin of type protoc?